### ISSUE TITLE
Mutations having @client fields are not returning server errors to the client.

### ISSUE THREAD

**Issue by cmnstmntmn** [apollographql/apollo-client#7565]
Hey, i think that the well known issue [like this here](https://github.com/apollographql/apollo-client/issues/7244) related to the absence of server errors, is caused by the presence of the `@client` directive inside the mutation response.

```js
MyClientSideResolvedField @client
```

Removing `@client` fields, is fixing the issue.

**Comment by benjamn** [comment]
@cmnstmntmn Just to make sure, you mean there's a server error for `ServerFieldOne` or `ServerFieldTwo`, but it's hidden when the `MyClientSideResolvedField @client` field is included in the mutation?

To implement the local field, are you using a local resolver function for that field (passed in `options.resolvers` to the `ApolloClient` constructor), or a field policy `read` function, or something else?

**Comment by cmnstmntmn** [comment]
@benjamn no, the `@client` field computes two other fields. They are not included in the mutation (as input).
They're just part of the mutation response

```gql
mutation SaveTicket(
  $ID: Int
  $LocationID: Int!
  $Type: TicketType!
  $Status: TicketStatus!
  $Urgency: TicketUrgency!
  $PatientID: Int! = 0
) {
  SaveTicket(
    input: {
      ID: $ID
      LocationID: $LocationID
      Type: $Type
      Status: $Status
      Urgency: $Urgency
      PatientID: $PatientID
    }
  ) {
    ID
    Type
    Status
    Patient {
      ID
      FirstName
      LastName
      FullName @client
    }
  }
}
```

The *FullName* resolver it's fine for queries; but it seems to be faulty with mutations.

As for resolvers, i'm passing them as options to the AplloClient constructor

```js
const apolloClient = new ApolloClient({
  cache,
  connectToDevTools: false,
  typeDefs,
  resolvers,
  link,
});
```

**Comment by timukasr** [comment]
Having same issue. One field in mutation has `@client` directive. When server returns error, eg: 
```{"errors":[{"message":"Something went wrong, please try again"}],"data":null}```
then this error is handled incorrectly by Apollo. It seems that it tries to read `data.mutationField`, and fails with error `Cannot read property 'mutationField' of null at new ApolloError`
